### PR TITLE
Add docs for PII management

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:25]: Can we make it more clear here why we are not anonimizing the tracker store in this version? 
Eg.: we want to retain a source of truth for pristine data for non-repudiation purposes. And probably also prediction..? (not sure)

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:30]: such a great overview:)

**Review by andreaklmn — CHANGES_REQUESTED** [review]
added a tiny comment:)

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx]: Addressed 👍🏻 
https://github.com/RasaHQ/rasa/pull/12505/commits/f1b891cdcec2108822c99ebc3dc629c88d44c5b4

File: docs/docs/pii-management.mdx
Comment (souvikg10) [inline docs/docs/pii-management.mdx:34]: > The Rasa agent will invoke the anonymization pipeline during each user message handling before saving the un-anonymized new events to the tracker store.

This line sounds a little contradicting 😕 we are running the anonymization pipeline and then saving un-anonymized events to the tracker store?

File: docs/docs/pii-management.mdx
Comment (souvikg10) [inline docs/docs/pii-management.mdx:175]: How is this different from `custom-logging-configuration`?

File: docs/docs/pii-management.mdx
Comment (souvikg10) [inline docs/docs/pii-management.mdx:181]: can this be also a list of rules?

Comment (ancalita) [reply docs/docs/pii-management.mdx:196]: No, this can only be a string and must match and `id` of the anonymization rule list in the `anonymization` section. So all rules are grouped only in that section.

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:34]: It's not contradictory, I mention in L26-29 that the tracker store remains single source of truth for pristine unanonymized data. The pipeline only anonymizes data streamed to specific topics consumed by Kafka consumers. Let me know how I should rephrase this to be clear.

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:190]: The logging file CLI option is used for configuring the  `logging` Python std library which was the dominant logging method we used in the codebase. However, in order to be able to hook the pipeline in PII leaking logs only (and not across all the possible logs), we chose to use `structlog` library instead which uses the [concept of processor chains](https://www.structlog.org/en/stable/getting-started.html#manipulating-log-entries-in-flight), we created our own processor (the anonymizer). In a nutshell, you can't use the logging file with structlog as far as I'm aware or it's for a separate ticket to work on how to enable this. Let me know if any of this needs to be clarified in the docs. I thought it's an implementation detail but the reality is that now the logs will be a mix of new structured logs and old logs with text messages.

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:32]: Addressed: https://github.com/RasaHQ/rasa/pull/12505/commits/89614a3ac29e11cec847a8e67ccc3a0407aad78f
I've clarified a few other assumptions and added some more detail in other parts, so re-requesting review.

File: docs/docs/glossary.mdx
Comment (ancalita) [inline docs/docs/glossary.mdx:22]: I saw a review note on this in my email notifications, but it doesn't seem to be in the GH UI anymore.
In any case, could you please pseudo-anonymization? Is that the same as replacing real PII with fake generated data? In which case I mention here as `artificial`, so I consider this to be a subset of anonymization. Let me know how I should rephrase to make it clearer.

File: docs/docs/pii-management.mdx
Comment (souvikg10) [reply docs/docs/pii-management.mdx:192]: I am not sure how many are using custom logging configuration anyway, perhaps we can move custom logging configurations to rasa pro and use endpoints.yml to harmonize it. but nevertheless it is a separate conversation. I dont see it contradicting the docs here so we are good

File: docs/docs/pii-management.mdx
Comment (souvikg10) [reply docs/docs/pii-management.mdx:36]: can we be explicit about the sequence of events that happens here.
- Agent invokes the anonymization pipeline
- Anonymized events are published onto the event broker
- un-anonymized events are then saved in the tracker store.

In the architecture overview, i think have the order of what happens to the events is important. 
Also are we only anonymising the UserMessage event or also BotUttered?

Comment (ancalita) [reply docs/docs/pii-management.mdx]: > can we be explicit about the sequence of events that happens here

Addressed: can we be explicit about the sequence of events that happens here

> are we only anonymising the UserMessage event or also BotUttered?

We are anonymizing UserUttered, BotUttered, SlotSet and EntitiesAdded, I've clarified the events in a subsection just below the Architecture Overview, although used their class type name rather than code reference name. Let me know which one would be clearer.

Comment (souvikg10) [reply docs/docs/pii-management.mdx]: I think class type is clear. makes sense. thank you. everything looks very clear to me 💯

Comment (ancalita) [reply docs/docs/pii-management.mdx]: Yay, that's great! Thank you for your thorough review ⚡ I will wait for Lena's review on Monday before merging this.

**Review by l2nguyen — APPROVED** [review]

File: docs/docs/command-line-interface.mdx
Comment (ancalita) [reply docs/docs/command-line-interface.mdx:52]: Nope, you can't set the log level separately for the two dependencies, both are covered by this env var.

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:145]: Nope, not possible at the moment, we only have support for the built-in Presidio entities. I'll see how I can clarify that regex is not supported.

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:131]: Good catch!

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:149]: Good point 👍🏻

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:145]: Addressed: https://github.com/RasaHQ/rasa/pull/12505/commits/ae10dcef4e1c8efe17e3e4f37b888af4506d3bd8

File: docs/docs/command-line-interface.mdx
Comment (andreaklmn) [inline docs/docs/command-line-interface.mdx:52]:
```suggestion
5. LOG_LEVEL_PRESIDIO: This is the specialized environment variable to configure log level only for Presidio, at the moment it handles log levels from `presidio_analyzer` and `presidio_anonymizer`.
```

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:27]:
```suggestion
retain a source of truth of pristine data for non-repudiation purposes. Learn how to secure your Tracker Store
```

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:40]:
```suggestion
1. The Rasa agent invokes the anonymization pipeline during each user message handling.
```

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:41]:
```suggestion
2. The anonymization pipeline runs through a sequence of anonymization rules that are applied to the new events.
```

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:42]:
```suggestion
3. The pipeline publishes the anonymized events to the Kafka topic that are mapped to the anonymization rule list in
```

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:44]:
```suggestion
4. The tracker store saves the original events that are not anonymized.
```

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:99]:
```suggestion
per anonymization pipeline.
```

Comment (andreaklmn) [reply docs/docs/pii-management.mdx:99]: I deleted this because it made us look like we are building explicitly for single language bots. But perhaps if we phrased it differently?

File: docs/docs/pii-management.mdx
Comment (andreaklmn) [inline docs/docs/pii-management.mdx:113]:
```suggestion
the HuggingFace model name and the spaCy model name, where the spaCy model would wrap the transformers NER model.
```

**Review by andreaklmn — CHANGES_REQUESTED** [review]

File: docs/docs/pii-management.mdx
Comment (ancalita) [reply docs/docs/pii-management.mdx:99]: We need to document this assumption because the pipeline won't be able to work with language switching at the moment. How about: `because the capability is currently not able to handle language switching by end users.`

Comment (ancalita) [reply docs/docs/pii-management.mdx]: Or maybe replace `because` with `therefore` 😄

**Review by andreaklmn — APPROVED** [review]

**Pull request merged**
